homeassistant/components/pyload/config_flow.py

```diff
@@ -0,0 +1,154 @@
+"""Config flow for pyLoad integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from aiohttp import CookieJar
+from pyloadapi.api import PyLoadAPI
+from pyloadapi.exceptions import CannotConnect, InvalidAuth, ParserError
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_NAME,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_SSL,
+    CONF_USERNAME,
+    CONF_VERIFY_SSL,
+)
+from homeassistant.core import DOMAIN as HOMEASSISTANT_DOMAIN, HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+from homeassistant.helpers.aiohttp_client import async_create_clientsession
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.issue_registry import IssueSeverity, async_create_issue
+
+from .const import DEFAULT_HOST, DEFAULT_NAME, DEFAULT_PORT, DOMAIN, ISSUE_PLACEHOLDER
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_NAME, description={"suggested_value": DEFAULT_NAME}): str,
+        vol.Required(
+            CONF_HOST, description={"suggested_value": "homeassistant.local"}
+        ): str,
+        vol.Required(CONF_PORT, default=DEFAULT_PORT): cv.port,
+        vol.Required(CONF_SSL, default=False): cv.boolean,
+        vol.Required(CONF_VERIFY_SSL, default=True): bool,
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+async def validate_input(hass: HomeAssistant, user_input: dict[str, Any]) -> None:
+    """Validate the user input and try to connect to PyLoad."""
+
+    session = async_create_clientsession(
+        hass,
+        user_input.get(CONF_VERIFY_SSL, True),
+        cookie_jar=CookieJar(unsafe=True),
+    )
+    host = user_input[CONF_HOST]
+    port = user_input[CONF_PORT]
+    protocol = "https" if user_input[CONF_SSL] else "http"
+    url = f"{protocol}://{host}:{port}/"
+    pyload = PyLoadAPI(
+        session,
+        api_url=url,
+        username=user_input[CONF_USERNAME],
+        password=user_input[CONF_PASSWORD],
+    )
+
+    await pyload.login()
+
+
+class PyLoadConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for pyLoad."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            try:
+                await validate_input(self.hass, user_input)
+            except (CannotConnect, ParserError):
+                errors["base"] = "cannot_connect"
+            except InvalidAuth:
+                errors["base"] = "invalid_auth"
+            except Exception:
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+            else:
```

Comment:
This is a common flow in config flows, otherwise you would need to have multiple `async_show_form` calls or make that a function, which I personally don't like because its making it a bit more verbose